autorag/nodes/queryexpansion/hyde.py

```diff
@@ -1,40 +1,40 @@
-import asyncio
-from typing import List
+from typing import List, Dict, Callable
 
-from llama_index.core.service_context_elements.llm_predictor import LLMPredictorType
+import pandas as pd
 
 from autorag.nodes.queryexpansion.base import query_expansion_node
-from autorag.utils.util import process_batch
 
 hyde_prompt = "Please write a passage to answer the question"
 
 
 @query_expansion_node
-def hyde(queries: List[str], llm: LLMPredictorType,
+def hyde(queries: List[str],
+         generator_func: Callable,
+         generator_params: Dict,
          prompt: str = hyde_prompt,
          batch: int = 16) -> List[List[str]]:
     """
     HyDE, which inspired by "Precise Zero-shot Dense Retrieval without Relevance Labels" (https://arxiv.org/pdf/2212.10496.pdf)
     LLM model creates a hypothetical passage.
     And then, retrieve passages using hypothetical passage as a query.
     :param queries: List[str], queries to retrieve.
-    :param llm: llm to use for hypothetical passage generation.
+    :param generator_func: Callable, generator functions.
+    :param generator_params: Dict, generator parameters.
     :param prompt: prompt to use when generating hypothetical passage
     :param batch: Batch size for llm.
         Default is 16.
     :return: List[List[str]], List of hyde results.
     """
-    # Run async query_decompose_pure function
-    tasks = [hyde_pure(query, llm, prompt) for query in queries]
-    loop = asyncio.get_event_loop()
-    results = loop.run_until_complete(process_batch(tasks, batch_size=batch))
+    full_prompts = []
+    for query in queries:
+        if prompt is "":
+            prompt = hyde_prompt
+        full_prompt = prompt + f"\nQuestion: {query}\nPassage:"
+        full_prompts.append(full_prompt)
+    full_prompts = list(
+        map(lambda x: (prompt if prompt else hyde_prompt) + f"\nQuestion: {x}\nPassage:", queries))
+    input_df = pd.DataFrame({"prompts": full_prompts})
+    result_df = generator_func(project_dir=None, previous_result=input_df, batch=batch, **generator_params)
```

Comment:
Suggest use generator_func.__wrapped__ instead.
No need to make fake project_dir or previous_result